extensions/data-plane/data-plane-google-bigquery/src/test/java/org/eclipse/edc/connector/dataplane/gcp/bigquery/pipeline/BigQueryDataSinkFactoryTest.java

```diff
@@ -0,0 +1,118 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.ExecutorService;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+
+public class BigQueryDataSinkFactoryTest {
+    private static final String TEST_PROJECT = "test-project";
+    private static final String TEST_DATASET = "test-dataset";
+    private static final String TEST_TABLE = "test-table";
+    private static final String TEST_OTHER_TYPE = "AnotherDataAddressType";
+    private static final String TEST_REQUEST_ID = "request-id";
+    private static final String TEST_PROCESS_ID = "process-id";
+    private static final String TEST_CUSTOMER_NAME = "customer-name";
+    private static final String TEST_SINK_SERVICE_ACCOUNT_NAME = "sinkAccount";
+    private BigQueryConfiguration bigQueryConfiguration = new BigQueryConfiguration(null, null, "testEndpoint", 0);
+    private TypeManager typeManager = mock();
+    private Monitor monitor = mock();
+    private ExecutorService executorService = mock();
+    private Vault vault = mock();
+    private IamService iamService = mock();
+
+    @BeforeEach
+    void setup() {
+        reset(monitor);
+        reset(typeManager);
+        reset(executorService);
+        reset(vault);
+        reset(iamService);
+
+        when(monitor.withPrefix(any(String.class))).thenReturn(monitor);
+    }
+
+    @Test
+    void testCanHandle() {
+        var provider = new BigQueryRequestParamsProvider();
+        var factory = new BigQueryDataSinkFactory(bigQueryConfiguration, executorService, monitor, vault, typeManager, provider, iamService);
+
+        var bqDataFlowRequest = getDataFlowRequest(BigQueryServiceSchema.BIGQUERY_DATA);
+
+        assertThat(factory.canHandle(bqDataFlowRequest)).isTrue();
+
+        var otherDataFlowRequest = getDataFlowRequest(TEST_OTHER_TYPE);
+
+        assertThat(factory.canHandle(otherDataFlowRequest)).isFalse();
+    }
+
+    @Test
+    void testValidateRequest() {
+        // TODO add tests if validateRequest body is implemented with specific tests.
+    }
+
+    @Test
+    void testCreateSink() {
+        var provider = new BigQueryRequestParamsProvider();
+        var factory = new BigQueryDataSinkFactory(bigQueryConfiguration, executorService, monitor, vault, typeManager, provider, iamService);
+
+        var bqDataFlowRequest = getDataFlowRequest(BigQueryServiceSchema.BIGQUERY_DATA);
+
+        var bqSink = factory.createSink(bqDataFlowRequest);
+        assertThat(bqSink).isNotNull();
+
+        var otherDataFlowRequest = getDataFlowRequest(TEST_OTHER_TYPE);
+
+        Throwable exception = assertThrows(GcpException.class, () -> factory.createSink(otherDataFlowRequest));
```

Comment:
nit: throughout the code base, we use `assertThatThrownBy(() -> {...}).hasMessage("some-message")`.